src/Task/RealMeSetupTask.php

```diff
@@ -206,20 +246,30 @@ private function isReadable($filename)
      * The service name and privacy realm need to be under 10 chars eg.
      * http://hostname.domain/serviceName/privacyRealm
      *
+     * @param string $forEnv
      * @return void
      */
-    private function validateEntityID()
+    private function validateEntityID($forEnv)
     {
         $entityId = $this->service->getSPEntityID();
 
         if (is_null($entityId)) {
-            $this->errors[] = _t('RealMeSetupTask.ERR_CONFIG_NO_ENTITYID', '', '', array('env' => $env));
+            $this->errors[] = _t(
+                self::class . '.ERR_CONFIG_NO_ENTITYID',
+                'No entityID specified for environment \'{env}\'. Specify this in your YML configuration, see the' .
+                ' module documentation for more details',
+                array('env' => $forEnv)
+            );
         }
 
         // make sure the entityID is a valid URL
         $entityId = filter_var($entityId, FILTER_VALIDATE_URL);
         if ($entityId === false) {
-            $this->errors[] = _t('RealMeSetupTask.ERR_CONFIG_ENTITYID', '', '',
+            $this->errors[] = _t(
+                self::class . '.ERR_CONFIG_ENTITYID',
+                'The Entity ID (\'{entityId}\') must be https, not be \'localhost\', and must contain a valid ' .
+                'service name and privacy realm e.g. https://my-realme-integration.govt.nz/p-realm/s-name',
+                '',
```

Comment:
This empty arg isn't needed. There are a few other occurrences of this but I won't keep listing them =)